csharp/src/Ice/OutputStream.cs

```diff
@@ -1405,7 +1408,7 @@ internal bool WriteSlicedData(SlicedData slicedData)
                 }
                 IceEndSlice(info.IsLastSlice); // TODO: can we check it's indeed the last slice?
             }
-            return firstSlice == false; // we wrote at least one slice
+            Debug.Assert(!firstSlice);
```

Comment:
This could be replaced with `Debug.Assert(slicedData.Slices.Count > 0);` at the top and we could get rid of this firstSlice bool which is only use for the assert.

Reply:
It's actually also used as parameter to the `IceStartSlice` call